### PR TITLE
feat: make output reflect user's choice of package manager

### PR DESCRIPTION
**Pull request by the-r3aper7**
### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [ ] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [ ] This message body should clearly illustrate what problems it solves.
- [ ] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [ ] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [ ] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpm changeset` and following the prompts. Changesets that add features should be `minor` and those that fix bugs should be `patch`. Please prefix changeset messages with `feat:`, `fix:`, or `chore:`.

### CODE REVIEW

**Comment by benmccann** [comment]
This seems like a good idea, but I wonder if we want to add an extra library for such a simple piece of functionality. Maybe it'd be preferable to just copy the source?

https://github.com/zkochan/packages/blob/main/which-pm-runs/index.js

**Comment by the-r3aper7** [comment]
yeah its a good idea, how should i provide the credits for the code from that package?

**Comment by benmccann** [comment]
You could just add a comment linking back to the original in the place where the code is added

**Comment by the-r3aper7** [comment]
@benmccann can you review this?

File: packages/create-svelte/utils.js [hunk 1]
```diff
@@ -49,3 +49,22 @@ export function copy(from, to, rename = identity) {
 export function dist(path) {
 	return fileURLToPath(new URL(`./dist/${path}`, import.meta.url).href);
 }
+
+// Thanks to https://github.com/zkochan/packages/tree/main/which-pm-runs for this code!
+export function getPackageManager() {
```
Comment (benmccann) [inline packages/create-svelte/utils.js:54]: we use snake_case for internal code and camelCase for external APIs
```suggestion
export function get_package_manager() {
```

File: packages/create-svelte/utils.js [hunk 2]
```diff
@@ -49,3 +49,22 @@ export function copy(from, to, rename = identity) {
...
+/** @param {string} userAgent */
+function pmFromUserAgent(userAgent) {
+	const pmSpec = userAgent.split(' ')[0];
+	const separatorPos = pmSpec.lastIndexOf('/');
+	const name = pmSpec.substring(0, separatorPos);
+	return {
+		name: name === 'npminstall' ? 'cnpm' : name,
+		version: pmSpec.substring(separatorPos + 1)
+	};
+}
```
Comment (benmccann) [inline packages/create-svelte/utils.js:70]:
```suggestion
function pm_from_user_agent(user_agent) {
	const pm_spec = user_agent.split(' ')[0];
	const separator_pos = pm_spec.lastIndexOf('/');
	const name = pm_spec.substring(0, separator_pos);
	return {
		name: name === 'npminstall' ? 'cnpm' : name,
		version: pm_spec.substring(separator_pos + 1)
	};
}
```

File: packages/create-svelte/utils.js [hunk 3]
```diff
@@ -49,3 +49,22 @@ export function copy(from, to, rename = identity) {
 export function dist(path) {
 	return fileURLToPath(new URL(`./dist/${path}`, import.meta.url).href);
 }
+
+// Thanks to https://github.com/zkochan/packages/tree/main/which-pm-runs for this code!
+export function getPackageManager() {
+	if (!process.env.npm_config_user_agent) {
+		return undefined;
+	}
+	return pmFromUserAgent(process.env.npm_config_user_agent);
```
Comment (benmccann) [inline packages/create-svelte/utils.js:58]:
```suggestion
	return pm_from_user_agent(process.env.npm_config_user_agent);
```

File: packages/create-svelte/utils.js (hunk 1 above)
Comment (the-r3aper7) [reply packages/create-svelte/utils.js]: okay i didn't knew that. ill make changes quickly.

File: packages/create-svelte/bin.js [hunk 4]
```diff
@@ -161,10 +163,10 @@ if (relative !== '') {
 	console.log(`  ${i++}: ${bold(cyan(`cd ${relative}`))}`);
 }
 
-console.log(`  ${i++}: ${bold(cyan('npm install'))} (or pnpm install, etc)`);
+console.log(`  ${i++}: ${bold(cyan(`${pkgManager} install`))}`);
```
Comment (benmccann) [inline packages/create-svelte/bin.js:166]:
```suggestion
console.log(`  ${i++}: ${bold(cyan(`${pkg_manager} install`))}`);
```

File: packages/create-svelte/bin.js [hunk 5]
```diff
@@ -161,10 +163,10 @@ if (relative !== '') {
 	console.log(`  ${i++}: ${bold(cyan(`cd ${relative}`))}`);
 }
 
-console.log(`  ${i++}: ${bold(cyan('npm install'))} (or pnpm install, etc)`);
+console.log(`  ${i++}: ${bold(cyan(`${pkg_manager} install`))}`);
 // prettier-ignore
 console.log(`  ${i++}: ${bold(cyan('git init && git add -A && git commit -m "Initial commit"'))} (optional)`);
-console.log(`  ${i++}: ${bold(cyan('npm run dev -- --open'))}`);
+console.log(`  ${i++}: ${bold(cyan(`${pkgManager} run dev -- --open`))}`);
```
Comment (benmccann) [inline packages/create-svelte/bin.js:169]:
```suggestion
console.log(`  ${i++}: ${bold(cyan(`${pkg_manager} run dev -- --open`))}`);
```

File: packages/create-svelte/utils.js [hunk 6]
```diff
@@ -49,3 +49,22 @@ export function copy(from, to, rename = identity) {
...
+
+// Thanks to https://github.com/zkochan/packages/tree/main/which-pm-runs for this code!
+export function get_package_manager() {
+	if (!process.env.npm_config_user_agent) {
+		return undefined;
+	}
+	return pm_from_user_agent(process.env.npm_config_user_agent);
+}
+
+/** @param {string} userAgent */
```
Comment (benmccann) [inline packages/create-svelte/utils.js:61]:
```suggestion
/** @param {string} user_agent */
```

**Review by benmccann — APPROVED** [review]

File: packages/create-svelte/utils.js (hunk 1 above)
Comment (benmccann) [reply packages/create-svelte/utils.js]: I made the changes, so it should be okay. You may want to test though. I just used the web UI and didn't test

**Review by Conduitry — CHANGES_REQUESTED** [review]
It looks like `get_package_manager()` is going to return a `{ name, version }` object, but we are treating the response as being just the name. Am I misreading something?

File: packages/create-svelte/utils.js [hunk 7]
```diff
@@ -49,3 +49,22 @@ export function copy(from, to, rename = identity) {
...
+
+/** @param {string} user_agent */
+function pm_from_user_agent(user_agent) {
+	const pm_spec = user_agent.split(' ')[0];
+	const separator_pos = pm_spec.lastIndexOf('/');
+	const name = pm_spec.substring(0, separator_pos);
+	return {
+		name: name === 'npminstall' ? 'cnpm' : name,
+		version: pm_spec.substring(separator_pos + 1)
+	};
```
Comment (benmccann) [inline packages/create-svelte/utils.js:69]:
```suggestion
	return name === 'npminstall' ? 'cnpm' : name;
```

**Comment by the-r3aper7** [comment]
yeah i'll make the changes i dont know how to test this that's why

File: packages/create-svelte/utils.js [hunk 8]
```diff
@@ -49,3 +49,19 @@ export function copy(from, to, rename = identity) {
...
+// Thanks to https://github.com/zkochan/packages/tree/main/which-pm-runs for this code!
+export function get_package_manager() {
+	if (!process.env.npm_config_user_agent) {
+		return undefined;
+	}
+	return pm_from_user_agent(process.env.npm_config_user_agent);
+}
+
+/** @param {string} user_agent */
+function pm_from_user_agent(user_agent) {
```
Comment (benmccann) [inline packages/create-svelte/utils.js:62]:
```suggestion
	const user_agent = process.env.npm_config_user_agent;
```

File: packages/create-svelte/utils.js [hunk 9]
```diff
@@ -49,3 +49,16 @@ export function copy(from, to, rename = identity) {
...
+export function get_package_manager() {
+	if (!process.env.npm_config_user_agent) {
+		return default_pkg_manager;
+	}
+	const user_agent = process.env.npm_config_user_agent;
+	const pm_spec = user_agent.split(' ')[0];
+	const separator_pos = pm_spec.lastIndexOf('/');
+	const name = pm_spec.substring(0, separator_pos);
+	return name === 'npminstall' ? 'cnpm' : name || default_pkg_manager;
+}
```
Comment (benmccann) [inline packages/create-svelte/utils.js:64]:
```suggestion
export package_manger = get_package_manager() || 'npm';

// Thanks to https://github.com/zkochan/packages/tree/main/which-pm-runs for this code!
function get_package_manager() {
	if (!process.env.npm_config_user_agent) {
		return undefined;
	}
	const user_agent = process.env.npm_config_user_agent;
	const pm_spec = user_agent.split(' ')[0];
	const separator_pos = pm_spec.lastIndexOf('/');
	const name = pm_spec.substring(0, separator_pos);
	return name === 'npminstall' ? 'cnpm' : name;
}
```

File: packages/create-svelte/bin.js [hunk 10]
```diff
@@ -4,11 +4,13 @@ import path from 'node:path';
...
 import { bold, cyan, grey, yellow } from 'kleur/colors';
 import { create } from './index.js';
-import { dist } from './utils.js';
+import { dist, get_package_manager } from './utils.js';
 
 const { version } = JSON.parse(fs.readFileSync(new URL('package.json', import.meta.url), 'utf-8'));
 let cwd = process.argv[2] || '.';
 
+const pkg_manager = get_package_manager();
+
```
Comment (benmccann) [inline packages/create-svelte/bin.js:13]:
```suggestion
```

**Comment by the-r3aper7** [comment]
guys you killing it let me make some changes too 🤣

File: packages/create-svelte/bin.js [hunk 11]
```diff
@@ -4,7 +4,7 @@ import path from 'node:path';
 import * as p from '@clack/prompts';
 import { bold, cyan, grey, yellow } from 'kleur/colors';
 import { create } from './index.js';
-import { dist } from './utils.js';
+import { dist, get_package_manager } from './utils.js';
```
Comment (benmccann) [inline packages/create-svelte/bin.js:7]:
```suggestion
import { dist, pkg_manager } from './utils.js';
```

File: packages/create-svelte/utils.js [hunk 12]
```diff
@@ -49,3 +49,16 @@ export function copy(from, to, rename = identity) {
 export function dist(path) {
 	return fileURLToPath(new URL(`./dist/${path}`, import.meta.url).href);
 }
+export package_manger = get_package_manager() || 'npm';
```
Comment (benmccann) [inline packages/create-svelte/utils.js:52]:
```suggestion

export const package_manger = get_package_manager() || 'npm';
```

File: packages/create-svelte/utils.js [hunk 13]
```diff
@@ -49,3 +49,17 @@ export function copy(from, to, rename = identity) {
 export function dist(path) {
 	return fileURLToPath(new URL(`./dist/${path}`, import.meta.url).href);
 }
+
+export const package_manger = get_package_manager();
+
+// Thanks to https://github.com/zkochan/packages/tree/main/which-pm-runs for this code!
+function get_package_manager() {
+	if (!process.env.npm_config_user_agent) {
+		return 'npm';
```
Comment (benmccann) [inline packages/create-svelte/utils.js:58]:
```suggestion
		return undefined;
```

File: packages/create-svelte/utils.js [hunk 14]
```diff
@@ -49,3 +49,17 @@ export function copy(from, to, rename = identity) {
 export function dist(path) {
 	return fileURLToPath(new URL(`./dist/${path}`, import.meta.url).href);
 }
+
+export const package_manger = get_package_manager();
```
Comment (benmccann) [inline packages/create-svelte/utils.js:53]:
```suggestion
export const package_manger = get_package_manager() || 'npm';
```

File: packages/create-svelte/utils.js [hunk 15]
```diff
@@ -49,3 +49,17 @@ export function copy(from, to, rename = identity) {
 export function dist(path) {
 	return fileURLToPath(new URL(`./dist/${path}`, import.meta.url).href);
 }
+
+export const package_manger = get_package_manager() || 'npm';
```
Comment (benmccann) [inline packages/create-svelte/utils.js:53]:
```suggestion
export const package_manager = get_package_manager() || 'npm';
```

File: packages/create-svelte/bin.js [hunk 16]
```diff
@@ -4,7 +4,7 @@ import path from 'node:path';
 import * as p from '@clack/prompts';
 import { bold, cyan, grey, yellow } from 'kleur/colors';
 import { create } from './index.js';
-import { dist } from './utils.js';
+import { dist, package_manger } from './utils.js';
```
Comment (benmccann) [inline packages/create-svelte/bin.js:7]:
```suggestion
import { dist, package_manager } from './utils.js';
```

File: packages/create-svelte/bin.js [hunk 17]
```diff
@@ -161,10 +161,10 @@ if (relative !== '') {
 	console.log(`  ${i++}: ${bold(cyan(`cd ${relative}`))}`);
 }
 
-console.log(`  ${i++}: ${bold(cyan('npm install'))} (or pnpm install, etc)`);
+console.log(`  ${i++}: ${bold(cyan(`${package_manger} install`))}`);
```
Comment (benmccann) [inline packages/create-svelte/bin.js:164]:
```suggestion
console.log(`  ${i++}: ${bold(cyan(`${package_manager} install`))}`);
```

File: packages/create-svelte/bin.js [hunk 18]
```diff
@@ -161,10 +161,10 @@ if (relative !== '') {
 	console.log(`  ${i++}: ${bold(cyan(`cd ${relative}`))}`);
 }
 
-console.log(`  ${i++}: ${bold(cyan('npm install'))} (or pnpm install, etc)`);
+console.log(`  ${i++}: ${bold(cyan(`${package_manager} install`))}`);
 // prettier-ignore
 console.log(`  ${i++}: ${bold(cyan('git init && git add -A && git commit -m "Initial commit"'))} (optional)`);
-console.log(`  ${i++}: ${bold(cyan('npm run dev -- --open'))}`);
+console.log(`  ${i++}: ${bold(cyan(`${package_manger} run dev -- --open`))}`);
```
Comment (benmccann) [inline packages/create-svelte/bin.js:167]:
```suggestion
console.log(`  ${i++}: ${bold(cyan(`${package_manager} run dev -- --open`))}`);
```

**Comment by benmccann** [comment]
thanks for catching that issue @Conduitry! I fixed it and made some other improvements before merging

**Pull request merged**